docs/providers/puppet.md

```diff
@@ -0,0 +1,74 @@
+# Puppet
+[Puppet][puppet] is a configuration management platform built by Puppet Labs.
+Magic Modules auto-generates a series of modules to allow Puppet to create/alter
+GCP resources.
+
+Each GCP product (Compute Engine, Cloud Storage, etc) has a module. A Puppet
+module is a collection of resources. Each resource corresponds one-to-one with
+a GCP resource.
+
+These modules are found on [Puppet Forge][puppet-forge]
+
+## Setting up a Puppet Development Environment
+### Installing Puppet
+The current version of Puppet (as of 05/2018) is 5.3.
+
+[Install instructions][puppet-install] for Linux are here.
+
+#### Verification
+```
+  puppet --version
+  > 5.3.0
+```
+
+### Placing Puppet Modules into proper location.
+Puppet modules can be installed from Supermarket using the following command:
+```
+   puppet module install google-gcompute
+```
+In many cases, you'll want to be running these modules from source.
+
+All Puppet modules are located in `~/.puppetlabs/etc/code/modules`. You'll have
+to place each module inside this folder using the naming convention
+`g<product name`. No underscores are necessary.
```

Comment:
nit, you probably meant `<product name>`

Reply:
Done!